### PR TITLE
tests: Add tests for cf

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: tests/dialects/test_cf.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    d = Subi.get(a, b)
+    e = Muli.get(a, b)
+
+    block0 = Block.from_ops([a, b, c])
+    block1 = Block.from_ops([d])
+
+    branch0 = ConditionalBranch.get(c, block0, [d], block1, [e])
+    assert type(branch0.then) is OpResult
+    assert type(branch0.then_arguments[0]) is OpResult
+    assert type(branch0.else_arguments[0]) is OpResult
```
Comment (webmiche) [inline tests/dialects/test_cf.py:31]: Maybe also test that the right block is on the right side?

I think the operation you are building here is not actually valid. There are block arguments for blocks that do not take arguments. I guess what this test shows is that our verifiers are not strong enough for the conditional branch. 

Maybe construct the blocks with block arguments or remove them form the conditional branch op...

**Review by webmiche — COMMENTED** [review]
With this PR, we actually cover more lines, but i feel that the second test in particular is not a good example for a conditional branch as it is ill-formed, AFAIU.

File: tests/dialects/test_cf.py (hunk 1 above)
Comment (georgebisbas) [reply tests/dialects/test_cf.py:31]: > Maybe also test that the right block is on the right side?
Could you elaborate?

Comment (webmiche) [reply tests/dialects/test_cf.py:26]: You could do something like `assert branch0.then is block0`.

Comment (georgebisbas) [reply tests/dialects/test_cf.py:26]: THis is still an OpResult, no, I managed to access through successors, is that right?

Comment (webmiche) [reply tests/dialects/test_cf.py:26]: Yes, you are right. Sorry, I was confused due to the naming of `then`. I feel this should be called `condition` or something.

File: tests/dialects/test_cf.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    # Operation to add these constants
+    c = Addi.get(a, b)
+    d = Subi.get(a, b)
+    e = Muli.get(a, b)
+
+    block0 = Block.from_arg_types([i32])
+    block1 = Block.from_arg_types([i32])
+
+    branch0 = ConditionalBranch.get(c, block0, [d], block1, [e])
+    assert type(branch0.then) is OpResult
```
Comment (math-fehr) [inline tests/dialects/test_cf.py:24]: I would probably replace that with `assert branch0.then is c` or something similar, to ensure the order in the constructor is correct as well!

**Review by math-fehr — APPROVED** [review]
I just added a small comment, but otherwise this is perfect!
Thanks!

File: tests/dialects/test_cf.py (hunk 2 above)
Comment (georgebisbas) [reply tests/dialects/test_cf.py:24]: Sure!

**Review by webmiche — APPROVED** [review]

**Pull request merged**